app/models/katello/concerns/smart_proxy_extensions.rb

```diff
@@ -456,14 +456,16 @@ def smart_proxy_service
       end
 
       def rhsm_url
+        proxy_url = setting('Registration', 'registration_url') || url
+
         # Since Foreman 3.1 this setting is set
         if (rhsm_url = setting(SmartProxy::PULP3_FEATURE, 'rhsm_url'))
           URI(rhsm_url)
           # Compatibility fall back
         elsif pulp_primary?
-          URI("https://#{URI.parse(url).host}/rhsm")
+          URI("https://#{URI.parse(proxy_url).host}/rhsm")
         elsif pulp_mirror?
-          URI("https://#{URI.parse(url).host}:8443/rhsm")
+          URI("https://#{URI.parse(proxy_url).host}:8443/rhsm")
```

Comment:
These code paths should be unused since Foreman 3.1.

Reply:
I grok from that only the initial `if` statement is ever triggered now, and is really not needed anymore. That, with this idea this function could be shortened to:

```
def rhsm_url

    url = setting('Registration', 'registration_url') || setting(SmartProxy::PULP3_FEATURE, 'rhsm_url')
    URI(url)

end
```

Reply:
Correct that only the initial `if` is triggered. Still, big :-1: from my side on this.

Reply:
@ekohl so what would be the proper fix  / update?